sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -75,57 +82,55 @@ object ScalaNativePluginInternal {
     }
   )
 
-  lazy val scalaNativeConfigSettings: Seq[Setting[_]] = Seq(
-    nativeTarget := interceptBuildException {
-      val cwd   = nativeWorkdir.value.toPath
-      val clang = nativeClang.value.toPath
+  def scalaNativeConfigSettings: Seq[Setting[_]] = Seq(
+    nativeTarget in nativeLink := interceptBuildException {
+      val cwd   = (nativeWorkdir in nativeLink).value.toPath
+      val clang = (nativeClang in nativeLink).value.toPath
       Discover.targetTriple(clang, cwd)
     },
     artifactPath in nativeLink := {
-      crossTarget.value / (moduleName.value + "-out")
+      (crossTarget in nativeLink).value / (moduleName.value + "-out")
     },
-    nativeWorkdir := {
-      val workdir = crossTarget.value / "native"
+    nativeWorkdir in nativeLink := {
+      val workdir = (crossTarget in nativeLink).value / "native"
       if (!workdir.exists) {
         IO.createDirectory(workdir)
       }
       workdir
     },
-    nativeConfig := {
-      val mainClass = selectMainClass.value.getOrElse {
-        throw new MessageOnlyException("No main class detected.")
-      }
-
-      val classpath =
-        fullClasspath.value.map(_.data.toPath).filter(f => Files.exists(f))
-      val maincls = mainClass.toString + "$"
-      val cwd     = nativeWorkdir.value.toPath
-      val clang   = nativeClang.value.toPath
-      val clangpp = nativeClangPP.value.toPath
-      val gc      = build.GC(nativeGC.value)
-      val mode    = build.Mode(nativeMode.value)
-
-      build.Config.empty
-        .withMainClass(maincls)
-        .withClassPath(classpath)
-        .withWorkdir(cwd)
-        .withClang(clang)
-        .withClangPP(clangpp)
-        .withTargetTriple(nativeTarget.value)
-        .withCompileOptions(nativeCompileOptions.value)
-        .withLinkingOptions(nativeLinkingOptions.value)
-        .withGC(gc)
-        .withMode(mode)
-        .withLinkStubs(nativeLinkStubs.value)
-        .withLTO(nativeLTO.value)
-        .withCheck(nativeCheck.value)
-        .withDump(nativeDump.value)
-        .withOptimize(Discover.optimize())
+    nativeConfig in nativeLink := {
```

Comment:
This `in nativeLink` is also unnecessary. The `nativeConfig.value` on the rhs of `:=` will delegate to `nativeConfig` in the project scope, and be used to see the value of `nativeConfig` in the configuration scope. Users can then configure it per-config without having to write `in nativeLink` themselves.